### DISCUSSION TITLE
How  to wire up Rx Tx on Creality Sermoon D1?

### DISCUSSION THREAD

**Discussion by LukaszKoc** [luc-github/ESP3D#847] [Questions]
I have trouble to find any instructions where to find Rx and TX paths or ideally goldPins on stock D1 matherBoard. 

Could it work to connect esp via  printer usb port?
Please help :)

**Comment by luc-github** [comment]
usb host is not yet available, that said espressif idf released some.drivers for several usb to serial chips so.I will work on the implementation soon, but this is for C3/S2/S3 only not for standard esp32, neither esp8266
Also I need to do some experiment to see if it work as expected, so it won't be for short term and  also it may need idf drivers be part as arduino package, so this could add delay of implementation

I will start on idf porting of esp3d first(esp3d-tft) and if ok , I will add usb  support on esp3d arduino if it make sense

**Reply by LukaszKoc** [reply]
Thank you for your anwer 
I have done some reading and I see now thats not easy task. 

Good luck and have fun with this project :)
